File: src/main/java/de/rubixdev/enchantedshulkers/enchantment/VoidEnchantment.java
package de.rubixdev.enchantedshulkers.enchantment;

import de.rubixdev.enchantedshulkers.Mod;
import de.rubixdev.enchantedshulkers.Utils;
import de.rubixdev.enchantedshulkers.config.WorldConfig;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.util.collection.DefaultedList;

import java.util.List;

public class VoidEnchantment extends ContainerEnchantment {
    public VoidEnchantment() {
        super(Mod.PORTABLE_CONTAINER_TARGET);
    }

    @Override
    protected boolean generate() {
        return WorldConfig.generateVoid();
    }

    public static boolean onItemPickup(ServerPlayerEntity player, ItemStack stack) {
        if (player.isCreative() && !WorldConfig.creativeVoid() || stack.isEmpty()) return false;
        Item item = stack.getItem();
        List<ItemStack> containerSlots = Utils.getContainers(player, Mod.VOID_ENCHANTMENT);
        for (ItemStack container : containerSlots) {
            DefaultedList<ItemStack> containerInventory = Utils.getContainerInventory(container, player);
            for (ItemStack innerStack : containerInventory) {
                if (innerStack.isOf(item)) {
                    stack.setCount(0);
                    return true;
                }
            }
        }
        return false;
    }
}

File: src/main/kotlin/de/rubixdev/enchantedshulkers/enchantment/VoidEnchantment.kt
package de.rubixdev.enchantedshulkers.enchantment

import de.rubixdev.enchantedshulkers.Mod
import de.rubixdev.enchantedshulkers.Utils
import de.rubixdev.enchantedshulkers.config.WorldConfig
import net.minecraft.item.ItemStack
import net.minecraft.server.network.ServerPlayerEntity

class VoidEnchantment : ContainerEnchantment(Mod.PORTABLE_CONTAINER_TARGET) {
    override fun generate() = WorldConfig.generateVoid

    companion object {
        @JvmStatic
        fun onItemPickup(player: ServerPlayerEntity, stack: ItemStack): Boolean {
            if (player.isCreative && !WorldConfig.creativeVoid || stack.isEmpty) return false
            val item = stack.item
            val containerSlots = Utils.getContainers(player, Mod.VOID_ENCHANTMENT)
            for (container in containerSlots) {
                val containerInventory = Utils.getContainerInventory(container, player)
                for (innerStack in containerInventory) {
                    if (innerStack.isOf(item)) {
                        stack.count = 0
                        return true
                    }
                }
            }
            return false
        }
    }
}
